### PR TITLE
LPS-197894 | Poshi Automation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# LPS-196842 - Product QA | Test Automation Creation

Hi! :)
This pr is corresponding to this ticket: [LPS-196842](https://liferay.atlassian.net/browse/LPS-196842) → [LPS-201194](https://liferay.atlassian.net/browse/LPS-201194)
https://github.com/liferay-echo/liferay-portal/pull/14295

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by rafaellajordao** [comment]
ci:test:echo

**Comment by rafaellajordao** [comment]
ci:test:echo

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 1]
```diff
@@ -25,6 +25,77 @@ definition {
...
+				key_position = 2,
+				locator1 = "xpath=(//*[@data-qa-id='row'][contains(@class,'list')])[${key_position}]",
+				value1 = "A Web Content Title");
+		}
+
+		task ("The user can select the Reverse Sort Direction button via the keyboard") {
+			AssertElementAccessible(locator1 = "ManagementBar#ORDER_ICON_GENERIC");
+
+			KeyPress(
+				locator1 = "ManagementBar#ORDER_ICON_GENERIC",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:62]: Use the previous element

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 2]
```diff
@@ -25,6 +25,77 @@ definition {
...
+				value1 = "\TAB");
+
+			KeyPress(
+				locator1 = "ManagementBar#ORDER_ICON_GENERIC",
+				value1 = "\ENTER");
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:68]: Modify locator LIST_ENTRY_X to `xpath=(//*[@data-qa-id='row'][contains(@class,'list')])[${key_position}]`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 3]
```diff
@@ -25,6 +25,77 @@ definition {
...
+			AssertTextEquals.assertPartialText(
+				key_position = 1,
+				locator1 = "xpath=(//*[@data-qa-id='row'][contains(@class,'list')])[${key_position}]",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:71]: LeiconList#LIST_ENTRY_X

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 4]
```diff
@@ -25,6 +25,77 @@ definition {
...
+				value1 = "A Web Content Title");
+
+			AssertTextEquals.assertPartialText(
+				key_position = 2,
+				locator1 = "xpath=(//*[@data-qa-id='row'][contains(@class,'list')])[${key_position}]",
+				value1 = "B Web Content Title");
+
+			KeyPress(
+				locator1 = "ManagementBar#ORDER_ICON_GENERIC",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:80]: Use the previous element

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 5]
```diff
@@ -1035,6 +1035,110 @@ definition {
...
+			AssertTextEquals.assertPartialText(
+				key_position = 2,
+				locator1 = "LexiconList#LIST_ENTRY_X",
+				value1 = "A Web Content Title");
+
+			AssertTextEquals(
+				locator1 = "Search#SEARCH_INFO",
+				value1 = "2 Results for \"web content\"");
+
+			Click.pauseClickAt(locator1 = "ManagementBar#FILTER_AND_ORDER");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:1134]: Click(locator1 = "ManagementBar#FILTER_AND_ORDER");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 6]
```diff
@@ -1035,6 +1035,110 @@ definition {
...
+		}
+
+		task ("When varying filters or searching if order by is applied") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = "test-site-name");
+
+			Search.searchCP(searchTerm = "Web");
+
+			ManagementBar.setFilterAndOrder(orderBy = "Title");
+
+			Click.clickNoMouseOver(locator1 = "ManagementBar#ORDER_ICON_GENERIC");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:1057]: Click(locator1 = "ManagementBar#ORDER_ICON_GENERIC");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 7]
```diff
@@ -25,6 +25,77 @@ definition {
...
+		task ("When searching for a term on the web content admin page") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = ${randomSiteName});
+
+			Search.searchCP(searchTerm = "Web");
+
+			ManagementBar.setFilterAndOrder(orderBy = "Title");
+
+			AssertTextEquals.assertPartialText(
+				key_position = 1,
+				locator1 = "xpath=(//*[@data-qa-id='row'][contains(@class,'list')])[${key_position}]",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:49]: LeiconList#LIST_ENTRY_X

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Hi @rafaellajordao, would you mind updating according to comments? Thanks

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 8]
```diff
@@ -25,6 +25,77 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-196842. The user can select the Reverse Sort Direction button via the keyboard."
+	@priority = 4
+	test RevertSortDirectionOfSearchResultsViaKeyboard {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:30]: Migrate to WebContentAdminWithKeyboard

**Comment by Tim-Cao** [comment]
@rafaellajordao, see changes https://github.com/liferay-echo/liferay-portal/pull/14442

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**